packages/wonder-blocks-dropdown/components/dropdown.js

```diff
@@ -0,0 +1,234 @@
+// @flow
+
+import * as React from "react";
+import ReactDOM from "react-dom";
+import {StyleSheet} from "aphrodite";
+import type {AriaProps, StyleType} from "@khanacademy/wonder-blocks-core";
+import {getClickableBehavior} from "@khanacademy/wonder-blocks-core";
+import DropdownCore from "./dropdown-core.js";
+import type {Item, DropdownItem} from "../util/types.js";
+import ActionItem from "./action-item.js";
+import OptionItem from "./option-item.js";
+import DropdownAnchor from "./dropdown-anchor.js";
+
+type Props = {|
+    ...AriaProps,
+
+    /**
+     * The child function, that takes in a function to open the dropdown and state and
+     * returns an Element which will become the opener element for the dropdown.
+     */
+    children: (
+        openDropdown: () => void,
+        state: ClickableState,
+    ) => React.Element<any>,
+
+    /**
+     * Can be used to override the state of the Dropdown by parent elemnents
+     */
+    opened?: boolean,
+
+    /**
+     * The items present in the Dropdown
+     */
+    menuItems: Array<Item> | Item,
+
+    /**
+     * A callback that returns items that are newly selected. Use only if this
+     * menu contains select items (and make sure selectedValues is defined).
+     */
+    onChange?: (selectedItems: Array<string>) => mixed,
+
+    /**
+     * The values of the items that are currently selected. Use only if this
+     * menu contains select items (and make sure onChange is defined).
+     */
+    selectedValues?: Array<string>,
+
+    /**
+     * Whether this menu should be left-aligned or right-aligned with the
+     * opener component. Defaults to left-aligned.
+     */
+    alignment?: "left" | "right",
+
+    /**
+     * Test ID used for e2e testing.
+     */
+    testId?: string,
+
+    /**
+     * Optional styling to add to the dropdown wrapper.
+     */
+    dropdownStyle?: StyleType,
+|};
+
+type ClickableState = {|
+    pressed: boolean,
+    hovered: boolean,
+    focused: boolean,
+|};
+
+type State = {|
+    /**
+     * Whether or not the dropdown is currently open.
+     */
+    open: boolean,
```

Comment:
I think we should use `opened` here as well.  I know we use `open` in other parts of the codebase but it's kind of ambiguous as to whether it's a verb or an adjective which is confusing.  Eventually we'd like to transition everything to using `opened`.